operator-framework-junit5/src/main/java/io/javaoperatorsdk/operator/junit/OperatorExtension.java

```diff
@@ -0,0 +1,272 @@
+package io.javaoperatorsdk.operator.junit;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.utils.Utils;
+import io.javaoperatorsdk.operator.Operator;
+import io.javaoperatorsdk.operator.api.ResourceController;
+import io.javaoperatorsdk.operator.api.config.BaseConfigurationService;
+import io.javaoperatorsdk.operator.api.config.ConfigurationService;
+import io.javaoperatorsdk.operator.api.config.Version;
+import io.javaoperatorsdk.operator.processing.ConfiguredController;
+import io.javaoperatorsdk.operator.processing.retry.Retry;
+
+import static io.javaoperatorsdk.operator.api.config.ControllerConfigurationOverrider.override;
+
+public class OperatorExtension
+    implements HasKubernetesClient,
+    BeforeAllCallback,
+    BeforeEachCallback,
+    AfterAllCallback,
+    AfterEachCallback {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(OperatorExtension.class);
+
+  private final KubernetesClient kubernetesClient;
+  private final ConfigurationService configurationService;
+  private final String namespace;
+  private final Operator operator;
+  private final boolean preserveNamespaceOnError;
+  private final boolean waitForNamespaceDeletion;
+
+  private OperatorExtension(
+      ConfigurationService configurationService, boolean preserveNamespaceOnError,
+      boolean waitForNamespaceDeletion) {
+    this.kubernetesClient = new DefaultKubernetesClient();
+    this.namespace = UUID.randomUUID().toString();
```

Comment:
Cannot we add more semantic name? Like name of the test method?

Reply:
yes, that's possible, let me try

Reply:
done